invenio_search/ext.py

```diff
@@ -31,35 +31,52 @@
 
 from pkg_resources import iter_entry_points, resource_filename, \
     resource_isdir, resource_listdir
+from werkzeug.utils import cached_property
 
 from . import config
 from .cli import index as index_cmd
+from .proxies import current_search_client
 from .utils import build_index_name
 
 
 class _SearchState(object):
     """Store connection to elastic client and regiter indexes."""
 
-    def __init__(self, app, entry_point_group=None, **kwargs):
+    def __init__(self, app,
+                 entry_point_group_mappings=None,
+                 entry_point_group_templates=None,
+                 **kwargs):
         """Initialize state.
 
         :param app: An instance of :class:`~flask.app.Flask`.
-        :param entry_point_group: The entrypoint group name to load plugins.
+        :param entry_point_group_mappings:
+            The entrypoint group name to load mappings.
+        :param entry_point_group_templates:
+            The entrypoint group name to load templates.
         """
         self.app = app
         self.mappings = {}
         self.aliases = {}
         self.number_of_indexes = 0
         self._client = kwargs.get('client')
+        self.entry_point_group_templates = entry_point_group_templates
+
+        if entry_point_group_mappings:
+            self.load_entry_point_group_mappings(entry_point_group_mappings)
 
-        if entry_point_group:
-            self.load_entry_point_group(entry_point_group)
+    @cached_property
+    def templates(self):
+        result = {}
```

Comment:
Even if it works, having a dict as the default value of `result` is strange. if `self.entry_point_group_templates` is set `result` will not be or behave like a dict.
```python
result = None
...
return {k: v for d in result for k, v in d.items()} if result else {}
```